pkg/producer/convertor/convertor.go

```diff
@@ -28,5 +29,10 @@ var ProtoSchemaConvertor = map[string]RegisterProtoSchema{}
 type IPFIXToKafkaConvertor interface {
 	// ConvertIPFIXMsgToFlowMsgs converts multiple data records in the IPFIX message
 	// to flow messages for any given proto schema.
-	ConvertIPFIXMsgToFlowMsgs(msg *entities.Message) []*protobuf.FlowMessage
+	ConvertIPFIXMsgToFlowMsgs(msg *entities.Message) []protoreflect.Message
+	// ConvertIPFIXRecordToFlowMsg converts a single IPFIX record to a kafka flow
+	// message for any given proto schema. This is useful for producers
+	// send single record at a time instead of a batch of records. Using the msg
+	// input parameter is optional depending on the fields in the proto schema.
+	ConvertIPFIXRecordToFlowMsg(msg *entities.Message, record entities.Record) protoreflect.Message
```

Comment:
why do we need msg as a parameter? In my view with msg this method is similar to `ConvertIPFIXMsgToFlowMsgs`. `ConvertIPFIXMsgToFlowMsgs` itself can also return a single `protoreflect.Message` for single record.

Reply:
You are right. Technically we do not need it. In an effort to reduce the duplicated code, I forced the parameter in there. Let me refactor it differently.